client/src/hooks/useNotifications.js

```diff
@@ -5,26 +5,41 @@ import { ResourcesPortalContext } from '../ResourcesPortalContext'
 
 export const useNotifications = () => {
   const { user, token } = useUser()
-  const { notifications, setNotifications } = React.useContext(
+  const { notificationCount, setNotificationCount } = React.useContext(
     ResourcesPortalContext
   )
 
-  React.useEffect(() => {
-    const fetchNotifications = async () => {
-      const notificationRequest = await api.user.notifications.list(
-        user.id,
-        token
-      )
-      if (notificationRequest.isOk && notificationRequest.response) {
-        const retrievedNotifications = notificationRequest.response.results
-        setNotifications(retrievedNotifications)
-      }
+  const fetchNotifications = async () => {
+    const notificationRequest = await api.user.notifications.list(
+      user.id,
+      token
+    )
+    if (notificationRequest.isOk && notificationRequest.response) {
+      const retrievedNotifications = notificationRequest.response.results
+      return retrievedNotifications
     }
 
-    if (user && !notifications) fetchNotifications()
-  })
+    return {}
+  }
+
+  const fetchNewNotifications = async () => {
+    // fetchs after the date saved in the user object
+    const notificationRequest = await api.user.notifications.filter(
+      user.id,
+      { created_at__gt: user.viewed_notifications_at },
+      token
+    )
+    if (notificationRequest.isOk && notificationRequest.response) {
+      const retrievedNotifications = notificationRequest.response.results
+      setNotificationCount(retrievedNotifications.length)
+      return retrievedNotifications
+    }
+
+    return {}
+  }
 
-  const getUnreadNotifications = () => {
+  const getUnreadNotifications = (notifications) => {
```

Comment:
should this return an empty array is there is no user?

Reply:
This function isn't really necessary, so I just refactored it away